pal03377/Athena-New
# Add evaluation mode running in batch mode

### Motivation and Context



### Description


- [x] Refactor BaseInfoHeader and extract the module select and config
- [x] Refactor `mode` route to `data/<data_mode>`
- [x] Add "view mode" selector: `module requests` and `evaluation mode` 
- [x] Define experiment
- [x] Configure module(s) and config(s) to run on
- [x] Conduct Experiment
    - [x] Show config (disabled) + display module type
    - [x] Sync submission views
    - [ ] ~~Rate feedback (correct, relevance, ...)~~ -> Follow-up PR (together with automatic evaluation)
    - [x] Send off data for training for each module (progress bar)
    - [ ] ~~Incremental learning mode with edit feedback! (for finetuning etc.)~~ -> Follow-up PR
    - [x] Batch mode (tutor feedback not editable)
    - [x] Import/Export (progress etc.)
- [x] Fix unreferenced feedback for evaluation data (0-0)

### Steps for Testing


1. Run playground
2. Check if module requests works with the refactored module selector (removed from base info header)
3. Go to evaluation mode
4. Define the experiment you want to conduct 
     - Lots of configuration possible
     - There is a training checkbox where you can optionally enable training data
     - Import/export of the experiment should work
5. Configure modules
     - Now define the configurations you want to use for the experiment
     - Import/export should work (only with the correct exercise type)
6. Conduct experiment
    - Press start -> Progress should be displayed correctly for each module
    - Press "Show config" to look at a modules config (in a dialog)
    - Switch submission to look at (should be in sync with all modules and the tutor feedback)
    - Import/Export basically at any time (might disable it for running modules in the future IDK)

In general, just play around with it and report any issues :)
Repeat for each exercise type (text + programming)


### Screenshots


#### Changes to the BaseInfoHeader and the Module Requests mode:
<img width="1309" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/b855beaf-3a8b-4917-a449-05d30bdbd49a">

#### Define Experiment
<img width="1328" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/ef90ad5a-2142-48d9-bf25-e5767d78721f">

#### Configure Modules
<img width="1337" alt="Screenshot 2023-08-13 at 14 18 43" src="https://github.com/ls1intum/Athena/assets/5898705/583e7202-bd5e-4959-87e9-b11fc65de2fc">

#### Conduct Experiment
**text** 
<img width="1331" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/010f6a02-c476-40c1-ac6a-e071ea920628">

**programming**
<img width="1333" alt="image" src="https://github.com/ls1intum/Athena/assets/5898705/fe195aa6-ecf7-4066-a961-de48ccab9388">
